main.go

```diff
@@ -58,10 +58,19 @@ func main() {
 	concurrentConnections := common.ComputeConcurrencyValue(runtime.NumCPU())
 	concurrentFilesLimit := computeConcurrentFilesLimit(maxFileAndSocketHandles, concurrentConnections)
 
-	err = ste.MainSTE(concurrentConnections, concurrentFilesLimit, 2400, azcopyAppPathFolder, azcopyLogPathFolder)
-	common.PanicIfErr(err)
+	// now we know how we want to start the STE, but create func to defer startup until after cmd line params are parsed
+	// TODO: reviewers, I'm not totally happy with the amount of indirection created by the use of this function.
+	//    Alternatives I thought of were:
+	//    1. Move more of the code that's here into root.go. Maybe that's an option...
+	//    2. Don't use a command line parameter for the cap on transfer rate, do it with an environment variable instead.
```

Comment:
I feel like an environment variable may actually be preferable to some users. I recall that the original intent of this was to "be a good neighbor" network wise. In a way, that sounded to me like users wanted a set-and-forget solution. This may actually also be preferable scripting wise.

I can see why it feels "wrong" though, since this may just be a single-transfer thing that the user wants.

Maybe the best solution is a mix of both. Infer from the environment variable unless the command line parameter is explicitly specified.

Reply:
Good question. Let's see what @zezha-msft says. Also, its worth noting that when we are called from tools like Storage Explorer is probably (slightly) easier for them to set a command line parameter.

Reply:
Good point @adreed-msft, but from my experience there are many users who launch the tool in simultaneous processes (not advised of course), so keeping it as a command line flag might be easier to use.

Reply:
I'm also leaning towards staying with the command line parameter. In my experience with other tooling, the first thing folks want to do with a feature like this is try it about 5 times with different speeds, and that's easier to do from the command line.  

Also, our current env vars are basically things which either we don't want in command line history (i.e. secrets) or things which we really wish the customer didn't need to think of at all.

Given the discussion so far, and the fact that having it on the command line worked well in FDT, I'm going to mark this conversation resolved.